### PR TITLE
Update doc for 4.0

### PR DESCRIPTION
**Pull request by VincentLanglet**
Already up to date with https://github.com/sonata-project/SonataAdminBundle/pull/6960

Closes: #6956

### CODE REVIEW

**Comment by franmomu** [comment]
Can't we do this on `3.x`?

**Comment by VincentLanglet** [comment]
> Can't we do this on `3.x`? (Maybe it's my fault for the issue description 😅 )

Typehint were added in 4.0, and some were modified in 4.0.

File: docs/cookbook/recipe_file_uploads.rst [hunk 1]
```diff
@@ -79,26 +79,20 @@ We then have the following methods in our ``Image`` class to manage file uploads
          */
         private $file;
 
-        /**
-         * @param UploadedFile $file
-         */
-        public function setFile(UploadedFile $file = null)
+        public function setFile(?UploadedFile $file = null)
```
Comment (jordisala1991) [inline docs/cookbook/recipe_file_uploads.rst:82]: return void

File: docs/cookbook/recipe_file_uploads.rst [hunk 2]
```diff
@@ -79,26 +79,20 @@ We then have the following methods in our ``Image`` class to manage file uploads
...
         {
             $this->file = $file;
         }
 
-        /**
-         * @return UploadedFile
-         */
-        public function getFile()
+        public function getFile(): UploadedFile
```
Comment (jordisala1991) [inline docs/cookbook/recipe_file_uploads.rst:87]: it can be null

File: docs/cookbook/recipe_file_uploads.rst (hunk 1 above)
Comment (VincentLanglet) [reply docs/cookbook/recipe_file_uploads.rst]: Done

File: docs/cookbook/recipe_file_uploads.rst (hunk 2 above)
Comment (VincentLanglet) [reply docs/cookbook/recipe_file_uploads.rst]: Done

**Review by jordisala1991 — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Pull request merged**